Model/Person/Tagging/Builder.php

```diff
@@ -37,57 +37,185 @@
+        $customer->setPhone($phone);
+        $customer->setPostCode($postCode);
+        $customer->setCountry($country);
+        $customer->setCustomerGroup($customerGroup);
+        $customer->setCustomerReference($customerReference);
+        $customer->setGender($gender);
+        if ($dateOfBirth !== null) {
+            $customer->setDateOfBirth(\DateTime::createFromFormat("Y-m-d", $dateOfBirth));
+        }
+
+        return $customer;
     }
 
     /**
      * Builds person from the current session / logged in user
      *
      * @param CurrentCustomer $currentCustomer
-     * @return AbstractPerson|null
+     * @return Customer|null
      */
     public function fromSession(CurrentCustomer $currentCustomer)
     {
         try {
             $customer = $currentCustomer->getCustomer();
+            $customerGroup = $this->getCustomerGroupName($customer);
+            $gender = $this->getGenderName($customer);
+            $customerReference = $this->getCustomerReference($currentCustomer);
+
             $person = $this->build(
                 $customer->getFirstname(),
-                $currentCustomer->getCustomer()->getLastname(),
-                $currentCustomer->getCustomer()->getEmail()
+                $customer->getLastname(),
+                $customer->getEmail(),
+                null,
+                null,
+                null,
+                $customerGroup,
+                $customer->getDob(),
+                $gender,
+                $customerReference
             );
 
             return $person;
         } catch (\Exception $e) {
             return null;
         }
     }
+
+    /**
+     * @param CustomerInterface $customer
+     * @return string
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    private function getCustomerGroupName(CustomerInterface $customer)
+    {
+        $groupId = (int)$customer->getGroupId();
+        $group = $this->groupRepository->getById($groupId);
+        return $group->getCode();
+    }
+
+    /**
+     * @param CustomerInterface $customer
+     * @return null|string
+     */
+    private function getGenderName(CustomerInterface $customer)
+    {
+        $gender = $customer->getGender();
+        switch ($gender) {
+            case self::GENDER_MALE_ID:
+                return self::GENDER_MALE;
+            case self::GENDER_FEMALE_ID:
+                return self::GENDER_FEMALE;
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * @param CurrentCustomer $currentCustomer
+     * @return string
+     */
+    private function getCustomerReference(CurrentCustomer $currentCustomer)
+    {
+        $customerReference = '';
+
+        try {
+            $customer = $currentCustomer->getCustomer();
+            $customerReference = $customer->getCustomAttribute(
+                NostoHelperData::NOSTO_CUSTOMER_REFERENCE_ATTRIBUTE_NAME
+            );
+
+            if (empty($customerReference)) {
```

Comment:
`$customer->getCustomAttribute` returns an object or null. Replace empty with `$customerReference === null`.

You can also invert the if and will look more readable.